sklearn/model_selection/_validation.py

```diff
@@ -961,6 +960,7 @@ def validation_curve(estimator, X, y, param_name, param_range, groups=None,
     X, y, groups = indexable(X, y, groups)
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
+    cv_iter = list(cv.split(X, y, groups))
```

Comment:
I think here the list can be safely removed without side effects as the iteration is only once and is based on splits...